src/pages/workspace/WorkspaceInitialPage.tsx

```diff
@@ -21,70 +22,69 @@ import useSingleExecution from '@hooks/useSingleExecution';
-import {policyDefaultProps, policyPropTypes} from './withPolicy';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type IconAsset from '@src/types/utils/IconAsset';
+import type {WithPolicyAndFullscreenLoadingProps} from './withPolicyAndFullscreenLoading';
 import withPolicyAndFullscreenLoading from './withPolicyAndFullscreenLoading';
 
-const propTypes = {
-    ...policyPropTypes,
+type WorkspaceMenuItem = {
+    translationKey: TranslationPaths;
+    icon: IconAsset;
+    action: () => void;
+    brickRoadIndicator?: ValueOf<typeof CONST.BRICK_ROAD_INDICATOR_STATUS>;
+};
 
-    /** All reports shared with the user (coming from Onyx) */
-    reports: PropTypes.objectOf(reportPropTypes),
+type WorkspaceInitialPageOnyxProps = {
+    /** All reports shared with the user */
+    reports: OnyxCollection<OnyxTypes.Report>;
 
     /** Bank account attached to free plan */
-    reimbursementAccount: ReimbursementAccountProps.reimbursementAccountPropTypes,
+    reimbursementAccount: OnyxEntry<OnyxTypes.ReimbursementAccount>;
 };
 
-const defaultProps = {
-    reports: {},
-    ...policyDefaultProps,
-    reimbursementAccount: {},
-};
+type WorkspaceInitialPageProps = WithPolicyAndFullscreenLoadingProps & WorkspaceInitialPageOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.INITIAL>;
 
-/**
- * @param {string} policyID
- */
-function openEditor(policyID) {
+function openEditor(policyID: string) {
     Navigation.navigate(ROUTES.WORKSPACE_SETTINGS.getRoute(policyID));
 }
 
-/**
- * @param {string} policyID
- */
-function dismissError(policyID) {
+function dismissError(policyID: string) {
     Navigation.goBack(ROUTES.SETTINGS_WORKSPACES);
     Policy.removeWorkspace(policyID);
 }
 
-function WorkspaceInitialPage(props) {
+function WorkspaceInitialPage({policyDraft, policy: policyProp, reports: reportsProp, policyMembers, reimbursementAccount}: WorkspaceInitialPageProps) {
     const styles = useThemeStyles();
-    const policy = props.policyDraft && props.policyDraft.id ? props.policyDraft : props.policy;
+    const policy = policyDraft?.id ? policyDraft : policyProp;
     const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
     const [isCurrencyModalOpen, setIsCurrencyModalOpen] = useState(false);
-    const hasPolicyCreationError = Boolean(policy.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD && policy.errors);
+    const hasPolicyCreationError = !!(policy?.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD && policy.errors);
     const waitForNavigate = useWaitForNavigation();
     const {singleExecution, isExecuting} = useSingleExecution();
 
     const {translate} = useLocalize();
     const {windowWidth} = useWindowDimensions();
 
-    const policyID = useMemo(() => policy.id, [policy]);
+    const policyID = useMemo(() => policy?.id ?? '', [policy]);
```

Comment:
Why do we need `useMemo` here? 
Looks like we can remove it

Reply:
Since it wasn't added in this PR, maybe I'd leave it as is